### PR TITLE
Add support to parse sub-aggregations from filter/nested aggregations

### PR DESCRIPTION
**Pull request by abhinav-nath**
Signed-off-by: Abhinav Nath <abhinavnath@ymail.com>

### Description
Add support to parse sub-aggregations from filter/nested aggregations.

### Issues Resolved
[Link](https://github.com/opensearch-project/opensearch-java/issues/197)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — APPROVED** [review]
I wonder whether we'd be better off separating the classes across tests. In this change `Product` serves multiple tests, so eventually it becomes a mess, doesn't it?

**Review by dblock — CHANGES_REQUESTED** [review]
We really need a USER_GUIDE similar to https://github.com/opensearch-project/opensearch-js/blob/main/USER_GUIDE.md here. Would you mind starting that as part of this PR with just documenting aggregations?

**Comment by abhinav-nath** [comment]
> We really need a USER_GUIDE similar to https://github.com/opensearch-project/opensearch-js/blob/main/USER_GUIDE.md here. Would you mind starting that as part of this PR with just documenting aggregations?

Sorry, was busy with work. I have added a USER_GUIDE.md with most of the basic operations (including aggregations). Please review and approve the PR now.

**Review by dblock — APPROVED** [review]
Looks great, thanks! Let's add a link to USER_GUIDE from README as well. Doesn't have to block this PR.

**Comment by abhinav-nath** [comment]
> > Looks great, thanks! Let's add a link to USER_GUIDE from README as well. Doesn't have to block this PR.
> 
> I have added the link to USER_GUIDE in the README.

@dblock Can we please merge this PR now? I really need it for a project. Thanks.

File: README.md
Comment (dblock) [inline README.md:26]: Add to TOC as well, please.

**Review by dblock — CHANGES_REQUESTED** [review]
The CHANGELOG verifier is complaining, need to add a line to that. Sorry about that ;(

File: README.md
Comment (abhinav-nath) [reply README.md:27]: Done. Sorry about so many commits. Hope you'll squash and merge :)

File: CHANGELOG.md
Comment (dblock) [inline CHANGELOG.md:8]: This one slipped in, bad copy-paste, remove.

File: README.md
Comment (dblock) [inline README.md:27]: Nit: Make it capital Code to match other titles, so "Sample Code" both in the TOC and in the actual section.

**Review by dblock — CHANGES_REQUESTED** [review]
Below ;)

Btw, if you want to make your commit look nicer after I squash it on merge, you can always squash and force push on the client. This can be a little tricky, I usually create a new branch off main, merge the change and force push it over, something like this (haven't actually tried):

```
git checkout main
git checkout -b subaggregations-squashed
git merge subaggregations --squash
git comit -s -m ....
git push origin -f subaggregations-squashed:subaggregations
```

File: CHANGELOG.md
Comment (abhinav-nath) [reply CHANGELOG.md]: Sorry, I just thought it was needed there. Fixed it.

File: README.md
Comment (abhinav-nath) [reply README.md]: > Nit: Make it capital Code to match other titles, so "Sample Code" both in the TOC and in the actual section.

Done.

**Review by dblock — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]
Thank you @abhinav-nath!

**Comment by abhinav-nath** [comment]
Thank you. Could you please add a "HACKTOBERFEST-ACCEPTED" label to this PR? Would be great for my dev profile 😊 
@dblock @VachaShah 

https://hacktoberfest.com/participation/#pr-mr-details

**Comment by abhinav-nath** [comment]
> Thank you. Could you please add a "HACKTOBERFEST-ACCEPTED" label to this PR? Would be great for my dev profile 😊 @dblock @VachaShah
> 
> https://hacktoberfest.com/participation/#pr-mr-details

Can someone do it please?

**Comment by dblock** [comment]
Done @abhinav-nath

**Comment by abhinav-nath** [comment]
> Done @abhinav-nath 

Thank you @dblock 🙏🏼

**Pull request merged**
